libyul/optimiser/VarNameCleaner.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <libyul/optimiser/VarNameCleaner.h>
+#include <libyul/AsmData.h>
+#include <libyul/Dialect.h>
+#include <algorithm>
+#include <cctype>
+#include <climits>
+#include <iterator>
+#include <string>
+#include <regex>
+
+using namespace yul;
+using namespace std;
+
+// ensure the following transforms
+// - input: a, a_1, a_1_2
+//	 output: a, a_1, a_2
+// - input: a, a_1, a_1_2, a_2
+//   output: a, a_1, a_2, a_3
+// - input: a_15, a_17
+//   output: a, a_2
+// - input: abi_decode_256
+//   output: abi_decode
+
+void VarNameCleaner::operator()(VariableDeclaration& _varDecl)
+{
+	for (TypedName& typedName: _varDecl.variables)
+		if (auto newName = makeCleanName(typedName.name.str()))
+			typedName.name = YulString{*newName};
+
+	ASTModifier::operator()(_varDecl);
+}
+
+void VarNameCleaner::operator()(Identifier& _identifier)
+{
+	if (auto newName = getCleanName(_identifier.name.str()))
```

Comment:
The distinction between `get` and `make` is not too big, I fear. What about `getNewlyAssignedName`? Oh and actually the coding style says to not use the `get` prefix.